pkg/yang/entry.go

```diff
@@ -726,6 +726,30 @@ func (e *Entry) Path() string {
 	return e.Parent.Path() + "/" + e.Name
 }
 
+// Namespace returns the YANG/XML namespace Value for e as mounted in the Entry
+// tree (e.g., as placed by grouping statements).
+//
+// Per RFC6020 section 7.12, the namespace on elements in the tree due to a
+// "uses" statement is that of the where the uses statement occurs, i.e., the
+// user, rather than creator (grouping) of those elements, so we follow the
+// usage (Entry) tree up to the parent before obtaining the (then adjacent) root
+// node for its namespace Value.
+func (e *Entry) Namespace() *Value {
+	// Make e the root parent entry
+	for ; e.Parent != nil; e = e.Parent {
```

Comment:
I think the following text found in [this note](https://tools.ietf.org/html/rfc6020#section-6.2.1) is the relevant section from RFC6020.

Unfortunately, I'm not certain whether "parent" in the below means the immediate or the ultimate parent node.

```
   o  All leafs, leaf-lists, lists, containers, choices, rpcs,
      notifications, and anyxmls defined (directly or through a uses
      statement) within a parent node or at the top level of the module
      or its submodules share the same identifier namespace.  This
      namespace is scoped to the parent node or module, unless the
      parent node is a case node.  In that case, the namespace is scoped
      to the closest ancestor node that is not a case or choice node.
```

What do you think?